src/styles/calendar.scss

```diff
@@ -0,0 +1,49 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+  max-width: $day-size * 7 + 6 * $gap;
+
+  @each $day, $size in $days-of-week {
+    &--start-day-#{$day} {
+      .calendar__day:first-child {
+        margin-left: $size * $start-day-size;
+      }
+    }
+  }
+
+  @each $length in $month-length {
+    &--month-length-#{$length} {
+      .calendar__day:nth-child(n + #{$length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    position: relative;
+    width: $day-size;
+    height: $day-size;
+    box-sizing: border-box;
```

Comment:
it s better to apply this property to all elements (use * selector)

When all elements have the same box-sizing value, their behavior becomes more predictable